clients/src/test/java/org/apache/kafka/common/network/SslVersionsTransportLayerTest.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.network;
+
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.kafka.common.config.SslConfigs;
+import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.common.security.TestSecurityConfig;
+import org.apache.kafka.common.security.auth.SecurityProtocol;
+import org.apache.kafka.common.utils.Java;
+import org.apache.kafka.common.utils.LogContext;
+import org.apache.kafka.common.utils.Time;
+import org.apache.kafka.test.TestUtils;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+/**
+ * Tests for the SSL transport layer.
+ * Checks different versions of the protocol usage on the server and client.
+ */
+@RunWith(value = Parameterized.class)
+public class SslVersionsTransportLayerTest {
+    private static final int BUFFER_SIZE = 4 * 1024;
+    private static final Time TIME = Time.SYSTEM;
+
+    private final String tlsServerProtocol;
+    private final String tlsClientProtocol;
+
+    @Parameterized.Parameters(name = "tlsServerProtocol={0},tlsClientProtocol={1}")
+    public static Collection<Object[]> data() {
+        List<Object[]> values = new ArrayList<>();
+        values.add(new Object[] {"TLSv1.2", "TLSv1.2"});
+        if (Java.IS_JAVA11_COMPATIBLE) {
+            values.add(new Object[] {"TLSv1.2", "TLSv1.3"});
+            values.add(new Object[] {"TLSv1.3", "TLSv1.2"});
+            values.add(new Object[] {"TLSv1.3", "TLSv1.3"});
+        }
+        return values;
+    }
+
+    public SslVersionsTransportLayerTest(String tlsServerProtocol, String tlsClientProtocol) {
+        this.tlsServerProtocol = tlsServerProtocol;
+        this.tlsClientProtocol = tlsClientProtocol;
+    }
+
+    /**
+     * Tests that connection success with the default TLS version.
+     */
+    @Test
+    public void testTLSDefaults() throws Exception {
```

Comment:
Nit: the test should be `testTlsDefaults` to match the naming convention.

Reply:
Fixed.